File: Backend/AstroGame.Shared/Models/Technologies/TechnologyUpgrade.cs
using System;

namespace AstroGame.Shared.Models.Technologies
{
    public abstract class TechnologyUpgrade
    {
        public Guid Id { get; set; }
        public Guid TechnologyId { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }

        public string HangfireJobId { get; set; }
        
        public virtual Technology Technology { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/technologies/technology_upgrade.dart
import 'package:astrogame_app/communications/converters/technology_upgrade_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';

abstract class TechnologyUpgrade {
  late Guid id;
  late Guid technologyId;
  late DateTime startTime;
  late DateTime endTime;
  late String hangfireJobId;

  TechnologyUpgrade();

  factory TechnologyUpgrade.fromJson(Map<String, dynamic> json) =>
      TechnologyUpgradeConverter().fromJson(json);
}
